SmartDeviceLink/SDLIAPSession.m

```diff
@@ -7,13 +7,20 @@
 #import "SDLStreamDelegate.h"
 #import "SDLTimer.h"
 
-NS_ASSUME_NONNULL_BEGIN
+#define IO_STREAMTHREAD_NAME            @ "com.smartdevicelink.iostream"
+#define STREAM_THREAD_WAIT_SECS         1.0
+#define STREAM_THREAD_WAIT_WRITE_SECS   5.0
 
 @interface SDLIAPSession ()
 
 @property (assign, nonatomic) BOOL isInputStreamOpen;
 @property (assign, nonatomic) BOOL isOutputStreamOpen;
-
+@property (assign, nonatomic) BOOL isDataSession;
+@property (nonatomic, copy) SessionSendHandler sendBlock;
+@property (nonatomic, strong) NSThread *ioStreamThread;
+@property (nonatomic, assign) BOOL isSendStarted;
```

Comment:
I'm not a huge fan of having so many state variables, would a change to a state machine make more sense here? Should we have multiple subclasses or adherents to a protocol to handle the differing data session / control session cases here?